nncf/quantization/algo.py

```diff
@@ -700,6 +709,25 @@ def __init__(self, target_model: NNCFNetwork,
             scheduler_cls = QUANTIZATION_SCHEDULERS.get("staged")
             self._scheduler = scheduler_cls(self, params)
 
+        if self._collect_compression_metrics:
+            self.metric_store = {}
+            quantizer_setup_type = self.quantization_config.get('quantizer_setup_type')
+            # These metrics are collected here and are updated when the method .statistics() is called
+            self.non_stable_metric_collectors = [NetworkQuantizationShareMetric(target_model, self.weight_quantizers,\
+                 self.non_weight_quantizers, quantizer_setup_type), MemoryСostMetric(target_model, self.weight_quantizers)]
+            # These metrics are collected once here and are not updated when the method .statistics() is called
+            self.stable_metric_collectors = [ShareEdgesQuantizedDataPath(target_model)]
+            self.update_metric_store(True)
+
+    def update_metric_store(self, _all: bool = False):
```

Comment:
All function arguments in Python are visible to users, and there is no such thing (and any feasible usage thereof) of "hidden" or "private" function arguments. The argument name should not be underscored.